nyc/bills.py

```diff
@@ -1,232 +1,359 @@
+    TIMEZONE = 'US/Eastern'
 
-    VOTE_OPTIONS = {'affirmative' : 'yes',
-                    'negative' : 'no',
-                    'conflict' : 'absent',
+    VOTE_OPTIONS = {'affirmative': 'yes',
+                    'negative': 'no',
+                    'conflict': 'absent',
                     'maternity': 'excused',
-                    'paternity' : 'excused',
+                    'paternity': 'excused',
                     'bereavement': 'excused',
-                    'non-voting' : 'not voting',
-                    'jury duty' : 'excused',
-                    'absent' : 'absent',
-                    'medical' : 'excused'}
+                    'non-voting': 'not voting',
+                    'jury duty': 'excused',
+                    'absent': 'absent',
+                    'medical': 'excused'}
+
+    # TO-DO: cq legislative session including 1996
+    SESSION_STARTS = (2014, 2010, 2006, 2004, 2002, 1998, 1996)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
 
-    SESSION_STARTS = (2014, 2010, 2006, 2002, 1996)
+        # Add required URL parameter
+        self.params = {'Token': TOKEN}
 
-    def sessions(self, action_date) :
-        for session in self.SESSION_STARTS :
-            if action_date >= datetime.datetime(session, 1, 1, 
-                                               tzinfo=pytz.timezone(self.TIMEZONE)) :
+
+    def sessions(self, action_date):
+        for session in self.SESSION_STARTS:
+            if action_date >= datetime.datetime(session, 1, 1,
+                                               tzinfo=pytz.timezone(self.TIMEZONE)):
                 return str(session)
 
 
+    def actions(self, matter_id):
+        for action in self.history(matter_id):
+            bill_action = {}
+
+            if not action['MatterHistoryActionName']:
+                continue
+
+            bill_action['action_description'] = action['MatterHistoryActionName']
+            bill_action['action_date'] = action['MatterHistoryActionDate']
+
+            responsible_org = action['MatterHistoryActionBodyName']
+
+            if responsible_org == 'City Council':
+                responsible_org = 'New York City Council'
+
+            elif responsible_org == 'Administration':
+                responsible_org = 'Mayor'
+
+            elif responsible_org == 'Town Hall Meeting':
+                continue
+
+            bill_action['responsible_org'] = responsible_org
+
+            if all(bill_action.values()):
+                action_date = self.toTime(bill_action['action_date']).date()
+                bill_action['action_date'] = action_date
+
+                # Strip trailing newline and carriage return chars
+                classification = ACTION_CLASSIFICATION[bill_action['action_description'].strip()]
+                bill_action['classification'] = classification
+
+                if all(action.get(k, None) for k in ['MatterHistoryEventId',
```

Comment:
I think it's slightly clearer to do something like this

```action.keys() >= {'MatterHistoryEventId', 'MatterHistoryRollCallFlag', 'MatterHistoryPassedFlag'}``

Reply:
This would just tell us whether the keys are present, no? We need to check that they're not null. However, the way I wrote it, this will fail if either flag is 0, which is not what we want.

How about something like...

```python
vote_fields = ('MatterHistoryEventId', 'MatterHistoryRollCallFlag', 'MatterHistoryPassedFlag')
if all([action[k] is not None for k in vote_fields]):
    # do vote stuff
```

?

Reply:
Much clearer what you want.